doc/reference/hv-make-options.rst

```diff
@@ -0,0 +1,199 @@
+.. _hypervisor-make-options:
+
+Hypervisor Makefile Options
+###########################
+
+The ACRN hypervisor source code provides a makefile to build the ACRN
+hypervisor binary and associated components.
+
+Assuming that you are at the top level of the ``acrn-hypervisor`` directory,
+you can run the ``make`` command to start the build. See :ref:`acrn_configuration_tool` for information about required input files.
+
+Build Options
+*************
+
+The following table shows ACRN-specific command-line options:
+
+.. list-table::
+   :widths: 33 77
+   :header-rows: 1
+
+   * - Option
+     - Description
+
+   * - ``BOARD``
+     - Required. Path to the board configuration file.
+
+   * - ``SCENARIO``
+     - Required. Path to the scenario configuration file.
+
+   * - ``RELEASE``
+     - Optional. Build a release version or a debug version. Valid values
+       are ``y`` for release version or ``n`` for debug version. (Default
+       is ``n``.)
+
+   * - ``hypervisor``
```

Comment:
@gvancuts Can you provide a description for "make doc"?

Reply:
ha.  How about:
```
Build the project's HTML documentation (using Sphinx), output to the ``_build/html`` folder.
```

Reply:
Looks good to me, except for the output folder, it's `build/doc` when you run `make doc` from the top-level directory. `_build/html` is when you call `make html` from within the `doc/` folder ;-)
```
Build the project's HTML documentation (using Sphinx), output to the ``_build/html`` folder.
```

Reply:
> Looks good to me, except for the output folder, it's `build/doc` when you run `make doc` from the top-level directory. `_build/html` is when you call `make html` from within the `doc/` folder ;-)
> 
> ```
> Build the project's HTML documentation (using Sphinx), output to the ``build/doc`` folder.
> ```

There is no "html" target in the top level Makefile -- it's "make doc" there.  We need to clean that up since the top level Makefile should simply call the doc/Makefile with the html target and not have doc building actions of its own.

Reply:
> There is no "html" target in the top level Makefile -- it's "make doc" there.

Yes, that's the additional target I want to see added to the list.

> We need to clean that up since the top level Makefile should simply call the doc/Makefile with the html target (which it does) and **not** change the default output folder for the doc build (remove the BUILDDIR parameter).

Hmm, no. I do not want a top-level `Makefile` target that puts stuff (by default) elsewhere than in the `build` directory. It's the same for all the other targets, if you call them from the subdirectory, they will be put their build artefacts in `hypervisor/build` (for example) but under `build/hypervisor` when called from the top-level `Makefile`.

Removing this `BUILDDIR` parameter would also make it awkward for the user in my opinion, as we would then tell him to:
1. Generate the documentation: `make doc`
2. Grab is from `doc/_build/html` instead of `build/doc` (which I find more intuitive).

Reply:
OK.  As you've done, documenting the "make doc" in the top-level directory works for me.